### PR TITLE
GEOPY-1310

### PR DESCRIPTION
**Pull request by domfournier**

### CODE REVIEW

File: curve_apps/driver.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    def params(self) -> BaseParameters:
+        """Application parameters."""
+        return self._params
+
+    @params.setter
+    def params(self, val: BaseParameters):
+        if not isinstance(val, BaseParameters):
+            raise TypeError("Parameters must be of type Parameters.")
+        self._params = val
+
```
Comment (MatthieuCMira) [inline curve_apps/driver.py:58]: Why do we want to duplicate the ui.json?

File: curve_apps/parts_connection/constants.py
Comment (MatthieuCMira) [inline curve_apps/parts_connection/constants.py:1]: Why do we want to store the uijson in a python file? 
I thought we need it as a ui.json file to launch it from GA?

File: curve_apps/parts_connection/driver.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            logger.info("Begin connecting labels ...")
+
+            vertices, cells, labels = self.get_connections()
+
+            logger.info("Process completed.")
+
+            if cells is None:
+                logger.info("No connections found.")
+                return
+
```
Comment (MatthieuCMira) [inline curve_apps/parts_connection/driver.py:66]: name as a function attribute?

File: curve_apps/parts_connection/driver.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            edges = Curve.create(
+                workspace=workspace,
+                name=name,
+                vertices=vertices,
+                cells=cells,
+                parent=parent,
+            )
+
+            if edges is not None:
```
Comment (MatthieuCMira) [inline curve_apps/parts_connection/driver.py:79]: Not a big deal: should we create a function for this guy, to keep run light

File: curve_apps/parts_connection/driver.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        """
+        max_distance = self.params.detection.max_distance
+
+        if max_distance is None:
+            max_distance = np.inf
+
+        path_list = []
+        out_labels = np.zeros_like(self.labels).astype("int32")
+
+        for value in tqdm(np.unique(self.labels)):
```
Comment (MatthieuCMira) [inline curve_apps/parts_connection/driver.py:120]: having a title in tqdm to explain the process?

File: tests/parts_connection_run_test.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        }
+    )
+
+    driver = PartsConnectionDriver(params)
+    driver.run()
+
+    with workspace.open():
+        edges = workspace.get_entity("test")[0]
+
+        assert len(edges.cells) == 27
```
Comment (MatthieuCMira) [inline tests/parts_connection_run_test.py:109]: isn't checking only for the length a bit light for a test?
we don't expect specific values?

File: tests/parts_connection_run_test.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
```
Comment (MatthieuCMira) [inline tests/parts_connection_run_test.py:136]: Same comment

File: tests/parts_connection_run_test.py [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    )
+    ifile.write_ui_json(str(tmp_path / "test_parts_connection"))
+    driver = PartsConnectionDriver(ifile)
+    driver.run()
+
+    with workspace.open():
+        edges = workspace.get_entity("square")[0]
+        assert edges is not None
+
+        assert any(child for child in edges.children if isinstance(child, FilenameData))
```
Comment (MatthieuCMira) [inline tests/parts_connection_run_test.py:163]: Agin, don't we accept a specific value?

**Review by MatthieuCMira — COMMENTED** [review]
No big comment. 


Just create more precise tests for assertion if possible.

File: curve_apps/parts_connection/constants.py
Comment (domfournier) [reply curve_apps/parts_connection/constants.py:1]: We need to store the base state somewhere, but I agree that we could use the file in assets

File: curve_apps/driver.py (hunk 1 above)
Comment (domfournier) [reply curve_apps/driver.py:90]: Done

File: curve_apps/parts_connection/constants.py
Comment (domfournier) [reply curve_apps/parts_connection/constants.py:1]: Done

File: curve_apps/parts_connection/driver.py (hunk 2 above)
Comment (domfournier) [reply curve_apps/parts_connection/driver.py]: Done

File: curve_apps/parts_connection/driver.py (hunk 3 above)
Comment (domfournier) [reply curve_apps/parts_connection/driver.py]: Done

File: curve_apps/parts_connection/driver.py (hunk 4 above)
Comment (domfournier) [reply curve_apps/parts_connection/driver.py]: I am acutally not sure how to do this. You have an example?

**Review by MatthieuCMira — APPROVED** [review]
Great!

I mark it as "approve"

Still, can you just increase your assertion in the test. 
It's very minor and you did it elsewhere, so it's going to be fast.

File: curve_apps/parts_connection/driver.py (hunk 4 above)
Comment (MatthieuCMira) [reply curve_apps/parts_connection/driver.py]: tqdm (self, iterable, desc= “Text You want”)

**Pull request merged**